pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1277,75 +1278,54 @@ func getMatchingTopologyPairs(pod *v1.Pod, nodeInfoMap map[string]*schedulercach
 			catchError(fmt.Errorf("node not found"))
 			return
 		}
-		nodeTopologyMaps := newTopologyPairsMaps()
 		for _, existingPod := range nodeInfo.PodsWithAffinity() {
-			affinity := existingPod.Spec.Affinity
-			if affinity == nil {
-				continue
-			}
-			for _, term := range GetPodAntiAffinityTerms(affinity.PodAntiAffinity) {
-				namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(existingPod, &term)
-				selector, err := metav1.LabelSelectorAsSelector(term.LabelSelector)
-				if err != nil {
-					catchError(err)
-					return
-				}
-				if priorityutil.PodMatchesTermsNamespaceAndSelector(pod, namespaces, selector) {
-					if topologyValue, ok := node.Labels[term.TopologyKey]; ok {
-						pair := topologyPair{key: term.TopologyKey, value: topologyValue}
-						nodeTopologyMaps.addTopologyPair(pair, existingPod)
-					}
-				}
-			}
-			if len(nodeTopologyMaps.podToTopologyPairs) > 0 {
-				appendTopologyPairsMaps(nodeTopologyMaps)
+			existingPodTopologyMaps, err := getMatchingAntiAffinityTopologyPairsOfPod(pod, existingPod, node)
+			if err != nil {
+				catchError(err)
+				return
 			}
+
+			appendTopologyPairsMaps(existingPodTopologyMaps)
 		}
 	}
 	workqueue.Parallelize(16, len(allNodeNames), processNode)
 	return topologyMaps, firstError
 }
 
-func getMatchingTopologyPairsOfExistingPod(newPod *v1.Pod, existingPod *v1.Pod, node *v1.Node) (*topologyPairsMaps, error) {
+// getMatchingAntiAffinityTopologyPairsOfPod is a helper to calculate on each existing pod of
+// given node, whether (1) it has PodAntiAffinity and (2) ALL AffinityTerms matches incoming pod
+func getMatchingAntiAffinityTopologyPairsOfPod(newPod *v1.Pod, existingPod *v1.Pod, node *v1.Node) (*topologyPairsMaps, error) {
+	if !targetPodMatchesAntiAffinityOfPod(existingPod, newPod) {
+		return nil, nil
```

Comment:
If this func can no longer return an error, consider removing error from the return type.

Reply:
yes, sort of. As old logic is replaced by `if !targetPodMatchesAntiAffinityOfPod(existingPod, newPod)` - where doesn't explicitly return error if it meets an error:

```go
    if err != nil {
        glog.Errorf("error in getting anti-affinity properties of Pod %v", pod.Name)
        return false
    }
```